include/cantera/thermo/EEDFTwoTermApproximation.h

```diff
@@ -0,0 +1,262 @@
+#define CT_EEDF_TWO_TERM_APPROXIMATION_H
+
+#include "cantera/base/ct_defs.h"
+#include "cantera/numerics/eigen_sparse.h"
+
+namespace Cantera
+{
+
+class PlasmaPhase;
+
+/**
+ *  EEDF solver options. Used internally by class EEDFTwoTermApproximation.
+ */
+class TwoTermOpt
+{
+public:
+    TwoTermOpt() = default;
+
+    double m_delta0 = 1e14; //!< Initial value of the iteration parameter
+    size_t m_maxn = 200; //!< Maximum number of iterations
+    double m_factorM = 4.0; //!< Reduction factor of error
+    size_t m_points = 150; //!< Number of points for energy grid
+    double m_rtol = 1e-5; //!< Relative tolerance of EEDF for solving Boltzmann equation
+    string m_growth = "temporal"; //!< String for the growth model (none, temporal or spatial)
+    double m_moleFractionThreshold = 0.01; //!< Threshold for species not considered in the Boltzmann solver but present in the mixture
+    string m_firstguess = "maxwell"; //!< String for EEDF first guess
+    double m_init_kTe = 2.0; //!< Initial electron mean energy
+
+}; // end of class TwoTermOpt
+
+//! Boltzmann equation solver for the electron energy distribution function based on
+//! the two-term approximation.
+//!
+//! @since New in %Cantera 3.2.
+//! @warning This class is an experimental part of %Cantera and may be changed without
+//!     notice.
+class EEDFTwoTermApproximation
+{
+public:
+    EEDFTwoTermApproximation() = default;
+
+    //! Constructor combined with the initialization function
+    /*!
+     * This constructor initializes the EEDFTwoTermApproximation object with everything
+     * it needs to start solving EEDF.
+     *
+     * @param s PlasmaPhase object that will be used in the solver calls.
+     */
+    EEDFTwoTermApproximation(PlasmaPhase& s);
+
+    virtual ~EEDFTwoTermApproximation() = default;
+
+    // compute the EEDF given an electric field
+    // CQM The solver will take the species to consider and the set of cross-sections
+    // from the PlasmaPhase object.
+    // It will write the EEDF and its grid into the PlasmaPhase object.
+    // Successful returns are indicated by a return value of 0.
+    int calculateDistributionFunction();
+
+    void setLinearGrid(double& kTe_max, size_t& ncell);
+
+    void setGridCache();
+
+    /**
+     * Options controlling how the calculation is carried out.
+     * @see TwoTermOpt
+     */
+    TwoTermOpt options;
+
+    vector<double> getGridEdge() const {
+        return m_gridEdge;
+    }
+
+    vector<double> getEEDFEdge() const {
+        return m_f0_edge;
+    }
+
+    double getElectronMobility() const {
+        return m_electronMobility;
+    }
+
+protected:
+    //! Pointer to the PlasmaPhase object used to initialize this object.
+    /*!
+     * This PlasmaPhase object must be compatible with the PlasmaPhase objects
+     * input from the compute function. Currently, this means that the 2
+     * PlasmaPhases have to have consist of the same species and elements.
+     */
+    PlasmaPhase* m_phase;
```

Comment:
I think this docstring is inconsistent with the implementation -- there doesn't appear to be a way to change the phase object after constructing this object (which is probably the right choice).